### PR TITLE
fix: Import local type from module and namespace

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**
```ts

declare module "foo2" {
    namespace Bar {
        interface I {
            a: string;
            b: number;
        }
    }

    namespace Baz {
        interface J {
            a: number;
            b: string;
        }
    }

    class Bar {
        item: Bar.I;
        constructor(input: Baz.J);
    }
}

let y: import("foo2").Bar.I = { a: "", b: 0 };
```

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs [hunk 1]
```diff
@@ -3053,16 +3053,22 @@ impl Analyzer<'_, '_> {
                             if let Some(vars) = exports.vars.get(sym).cloned() {
                                 return Ok(vars);
                             }
+
+                            for (id, tys) in exports.private_types.iter() {
+                                if id.sym().eq(sym) {
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs:3058]: Please use `==` instead

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs:3058]: Is there a particular reason other than readability?

Comment (sunrabbit123) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs:3058]:
```suggestion
                                if *id.sym() == *sym {
```

File: crates/stc_ts_file_analyzer/src/analyzer/scope/mod.rs [hunk 2]
```diff
@@ -2247,6 +2252,16 @@ impl Expander<'_, '_, '_> {
 
                             Type::Mapped(m) => {}
 
+                            Type::Namespace(..) => {
+                                dbg!(&t);
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/scope/mod.rs:2256]: Please remove this

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/scope/mod.rs:2256]: !! my mistake
I'll fix it right away
thank you

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**